lib/iris/fileformats/netcdf.py

```diff
@@ -780,6 +784,13 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
                     'cf_patch')
                 warnings.warn(msg)
 
+        # Perform any user registered callback function.
+        # As the NetCDF saver commits to disk as it processes the cube and
+        # its metadata, it is not possible to skip or ignore the derived
+        # NetCDF representation of a cube as with other file format savers.
```

Comment:
:see_no_evil: ... :hear_no_evil: ... hmmm indeed. Shame huh?

The callbacks only really make sense for `PP` and `GRIB` and not so much for `NetCDF` given the way that the savers been currently implemented.